app/models/account.rb

```diff
@@ -112,7 +112,7 @@ class Account < ApplicationRecord
   scope :matches_domain, ->(value) { where(arel_table[:domain].matches("%#{value}%")) }
   scope :without_unapproved, -> { left_outer_joins(:user).remote.or(left_outer_joins(:user).merge(User.approved.confirmed)) }
   scope :searchable, -> { without_unapproved.without_suspended.where(moved_to_account_id: nil) }
-  scope :discoverable, -> { searchable.without_silenced.where(discoverable: true).left_outer_joins(:account_stat) }
+  scope :discoverable, -> { searchable.without_silenced.where(discoverable: true).joins(:account_stat) }
```

Comment:
This means that accounts that don't yet have an `account_stat` row (e.g. 0 posts, following, followers) will not be returned. That is actually probably fine in practice, but it is different, and I'm not really sure how this would fix a HTTP 500 error.

Reply:
Because of another recent backported change (#27306), the generated SQL was a mess with two joins on `account_stat`, which caused the ambiguous query error. This PR fixes it. It's true that this makes accounts with 0 (known) posts or followers not appear, but I think this is fine in the profile directory.